### ISSUE TITLE
"git commit --amend" feature in github for mac

### ISSUE THREAD

**Issue by zrrtcs** [baldwindavid/github-for-mac-issues#30]
some of us who sometimes carelessly done commit need this. thanks.

ps: please do not leave a comment in this issue thread here only for "+1"

**Comment by paulocheque** [comment]
+1

**Comment by thomasmery** [comment]
+1

**Comment by IngwiePhoenix** [comment]
Absolutely need that O.O!
+2

**Comment by stickel** [comment]
:+1:

**Comment by mrmurphy** [comment]
+1

**Comment by borisdivjak** [comment]
+1

**Comment by jiserra** [comment]
+1

**Comment by wowthatisrandom** [comment]
+1

**Comment by josh18** [comment]
+1

**Comment by olutsenko** [comment]
+1

**Comment by nfarina** [comment]
Can't the same effect be accomplished via the "Edit" button on the last "unsynced commit"?

![screen shot 2014-05-08 at 7 03 04 pm](https://cloud.githubusercontent.com/assets/117280/2923699/199e149a-d71e-11e3-9331-124d986feea7.png)

**Comment by stickel** [comment]
> Can't the same effect be accomplished via the "Edit" button on the last "unsynced commit"?

Yes, but doesn't help if you have sync'd all your commits. The added steps of having to hit the 'edit' button, make sure the new changes you want aren't overridden, then running commit again are more tedious than, say, a checkbox option to `commit --amend`

**Comment by IngwiePhoenix** [comment]
Am 11.05.2014 um 02:27 schrieb Stickel notifications@github.com:

> Can't the same effect be accomplished via the "Edit" button on the last "unsynced commit"?
> 
> Yes, but doesn't help if you have sync'd all your commits. The added steps of having to hit the 'edit' button, make sure the new changes you want aren't overridden, then running commit again are more tedious than, say, a checkbox option to commit —amend
> 
> Not to mention that it also looks cleaner in your commit history.

**Comment by nfarina** [comment]
> > Can't the same effect be accomplished via the "Edit" button on the last "unsynced commit"?
> 
> Yes, but doesn't help if you have sync'd all your commits.

Well this is kind of a different feature request - modifying a commit post-sync would require a force push, which has new implications.

**Comment by MartinMekk** [comment]
+1

**Comment by knightq** [comment]
+1

**Comment by elving** [comment]
+1

**Comment by msanders** [comment]
+1

**Comment by arena** [comment]
+100

**Comment by arena** [comment]
In the meantime, if autocorrect changes the word "svg" to "sag" and makes you sound weird. You can go to the terminal and use these fun commands like I just did (proceed with caution):
git commit --amend -m "New commit message"
git push -f

**Comment by lazutkin** [comment]
+100500

**Comment by asm** [comment]
+1

**Comment by Andrew-Morozko** [comment]
+1

**Comment by adamwoodnz** [comment]
+1

**Comment by 0x0584** [comment]
+1

**Comment by mspangdal** [comment]
+1

**Comment by v1co0** [comment]
+ 1 TRILLION

**Comment by damiendda** [comment]
+ 1

**Comment by pbrueske** [comment]
+1

**Comment by chriswalz** [comment]
+1

**Comment by KayoticSully** [comment]
👍

**Comment by stianhoiland** [comment]
+1

Why isn't this implemented yet?

EDIT: I guess it's because of the undo commit feature which works kinda the same, though amending doesn't require reselecting lines/chunks.

**Comment by javiergranados** [comment]
+1

**Comment by mattrc** [comment]
+1

**Comment by koskoci** [comment]
+1

**Comment by drinkthestars** [comment]
+1

**Comment by Pipinha** [comment]
+1

**Comment by zrrtcs** [comment]
PLEASE STOP POSTING +1 here.
This is a very old issue been 7 years since reported and I have not checked since then as I am using other git client now.

**Comment by koffsky** [comment]
> PLEASE STOP POSTING +1 here.
> This is a very old issue been 7 years since reported and I have not checked since then as I am using other git client now.

+1

**Comment by arena** [comment]
> PLEASE STOP POSTING +1 here.
> This is a very old issue been 7 years since reported and I have not checked since then as I am using other git client now.

+1
